src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -298,6 +298,7 @@
   private boolean stopping = false;
   private boolean stopped = true;
   private Thread prometheusMetricsThread;
+  private boolean useDenyList = true;
```

Comment:
This can be final, no? Move it up if so.

Reply:
fixed